src/main/java/io/dataspaceconnector/common/net/HttpService.java

```diff
@@ -161,10 +166,15 @@ public Response get(final URL target, final HttpArgs args) throws IOException {
         Utils.requireNonNull(args, ErrorMessage.HTTP_ARGS_NULL);
 
         final var urlBuilder = createUrlBuilder(target);
+        HttpUrl targetUrl = toUrl(target);
+        var queryKeys = targetUrl.queryParameterNames();
 
         if (args.getParams() != null) {
             for (final var key : args.getParams().keySet()) {
-                urlBuilder.addQueryParameter(key, args.getParams().get(key));
+                // makes sure the passed queries do not have the same key as queries in the target
```

Comment:
Please remove the comment, this is already clear from the JavaDoc above the method.